src/TabBar.tsx

```diff
@@ -1,130 +1,142 @@
+        scrollViewRef.current.scrollTo({
+          x: getScrollAmount(),
+          animated: true,
+        })
       }
     }
-  }
-
-  getScrollAmount = () => {
-    const { currentIndex } = this.props;
-    const position = currentIndex;
-    const pageOffset = 0;
-
-    const containerWidth = WindowWidth;
-    const tabWidth = this._tabsMeasurements[position].width;
-    const nextTabMeasurements = this._tabsMeasurements[position + 1];
-    const nextTabWidth =
-      (nextTabMeasurements && nextTabMeasurements.width) || 0;
-    const tabOffset = this._tabsMeasurements[position].left;
-    const absolutePageOffset = pageOffset * tabWidth;
-    let newScrollX = tabOffset + absolutePageOffset;
-
-    newScrollX -=
-      (containerWidth -
-        (1 - pageOffset) * tabWidth -
-        pageOffset * nextTabWidth) /
-      2;
-    newScrollX = newScrollX >= 0 ? newScrollX : 0;
-
-    const rightBoundScroll = Math.max(
-      this._tabContainerMeasurements.width - containerWidth,
-      0
-    );
-
-    newScrollX = newScrollX > rightBoundScroll ? rightBoundScroll : newScrollX;
-    return newScrollX;
-  }
-
-  onTabContainerLayout = (e: LayoutChangeEvent) => {
-    this._tabContainerMeasurements = e.nativeEvent.layout;
-  }
-
-  onTabLayout = (key: number) => (ev: LayoutChangeEvent) => {
-    const { x, width, height } = ev.nativeEvent.layout;
-    this._tabsMeasurements[key] = {
-      left: x,
-      right: x + width,
-      width,
-      height
-    };
-  }
-
-  renderTab = (section: SectionListData<any>, key: number) => {
-    const { renderTab, onPress, currentIndex } = this.props;
-    const isActive: boolean = currentIndex === key;
-
-    return (
-      <TouchableOpacity
-        onPress={() => onPress(key)}
-        key={key}
-        onLayout={this.onTabLayout(key)}
+  }, [currentIndex, getScrollAmount])
+
+  const onLayout = (e: LayoutChangeEvent) => setTabContainerMeasurements(e.nativeEvent.layout)
+
+  const onTabLayout = useCallback(
+    (key: number) => (ev: LayoutChangeEvent) => {
+      const { x, width, height } = ev.nativeEvent.layout
+      setTabsMeasurements((oldTabsMeasurements) => ({
+        ...oldTabsMeasurements,
+        [key]: {
+          left: x,
+          right: x + width,
+          width,
+          height,
+        },
+      }))
+    },
+    [],
+  )
+
+  const renderLocalTab = useCallback(
+    (section: SectionListData<any>, key: number) => {
+      const isActive = currentIndex === key
+
+      return (
+        <TouchableOpacity {...{ key }} onPress={() => onPress(key)} onLayout={onTabLayout(key)}>
+          {renderTab({ isActive, ...section })}
+        </TouchableOpacity>
+      )
+    },
+    [currentIndex, onPress, onTabLayout, renderTab],
+  )
+
+  return (
+    <View style={s.view}>
+      <ScrollView
+        horizontal
+        contentContainerStyle={styles.row}
+        showsHorizontalScrollIndicator={false}
+        ref={scrollViewRef as React.RefObject<ScrollView>}
```

Comment:
I cast scrollViewRef because of this TS error : 
```
TS2769: No overload matches this call.   Overload 1 of 2, '(props: ScrollViewProps | Readonly<ScrollViewProps>): ScrollView', gave the following error.     Type 'MutableRefObject<ScrollView | undefined>' is not assignable to type 'LegacyRef<ScrollView> | undefined'.       Type 'MutableRefObject<ScrollView | undefined>' is not assignable to type 'RefObject<ScrollView>'.         Types of property 'current' are incompatible.           Type 'ScrollView | undefined' is not assignable to type 'ScrollView | null'.             Type 'undefined' is not assignable to type 'ScrollView | null'.   Overload 2 of 2, '(props: ScrollViewProps, context: any): ScrollView', gave the following error.     Type 'MutableRefObject<ScrollView | undefined>' is not assignable to type 'LegacyRef<ScrollView> | undefined'.
```

Reply:
Normally, using the following syntax is not supposed to generate errors. If so, check your linters options and typescript version.

```TSX
  const ref = useRef<ScrollView>(null)

  const Toto = () => <ScrollView ref={ref} />
```